### PR TITLE
partition_manager: Update documentation

### PR DESCRIPTION
**Pull request by de-nordic**
Update documentation to include information on changes to add_region
macro.

Signed-off-by: Dominik Ermel <dominik.ermel@nordicsemi.no>

### CODE REVIEW

**Review by shantha-14 — APPROVED** [review]

**Review by mbolivar-nordic — COMMENTED** [review]
No objections, needs tech review

File: scripts/partition_manager/partition_manager.rst [hunk 1]
```diff
@@ -507,7 +509,8 @@ The Partition Manager supports partitions in the external flash memory through t
 Any placeholder partition can specify that it should be stored in the external flash memory region.
 External flash memory regions always use the *start_to_end* placement strategy.
 
-To store partitions in the external flash memory, you must choose a value for the ``nordic,pm-ext-flash`` property in the devicetree.
+To store partitions in the external flash memory, you can choose a value for the ``nordic,pm-ext-flash`` property in the devicetree,
+or directly use external flash DTS node label as DEVICE.
```
Comment (FrancescoSer) [inline scripts/partition_manager/partition_manager.rst:513]:
```suggestion
To store partitions in the external flash memory, you can either choose a value for the ``nordic,pm-ext-flash`` property in the devicetree, or directly use an external flash DTS node label as ``DEVICE``.
```

Is the 2nd part of the sentence supposed to mean that the " external flash DTS node label" should be used as "DEVICE" or that the  external flash DTS node should be labelled as "DEVICE"?

File: scripts/partition_manager/partition_manager.rst [hunk 2]
```diff
@@ -527,6 +530,14 @@ See :ref:`ug_bootloader_external_flash` for more details on using external flash
 
 .. _pm_build_system:
 
+Flash partitions requre device drivers to be enabled for them. The ``DEFAULT_DRIVER_KCONFIG``
+parameter should be set to a Kconfig option that enables the driver during compilation.
+When a driver for a partition is not enabled, such partition will not be included into
+partition man and will not be accessible at run-time.
```
Comment (FrancescoSer) [inline scripts/partition_manager/partition_manager.rst:536]:
```suggestion
When a driver for a partition is not enabled, such partition will not be included into partition manager and will not be accessible at run-time.
```

File: scripts/partition_manager/partition_manager.rst [hunk 3]
```diff
@@ -527,6 +530,14 @@ See :ref:`ug_bootloader_external_flash` for more details on using external flash
...
+kconfig:option:`CONFIG_PM_OVERRIDE_EXTERNAL_DRIVER_CHECK` can be used to override the behaviour
+and prevent compile-time check for external flash devices; this is useful when external device
+does not have Kconfig conroller driver.
```
Comment (FrancescoSer) [inline scripts/partition_manager/partition_manager.rst:539]:
```suggestion
You can use kconfig:option:`CONFIG_PM_OVERRIDE_EXTERNAL_DRIVER_CHECK` to override the behaviour
and prevent compile-time check for external flash memory devices.
This is useful when the external device does not have a Kconfig controller driver.
```

File: scripts/partition_manager/partition_manager.rst [hunk 4]
```diff
@@ -527,6 +530,14 @@ See :ref:`ug_bootloader_external_flash` for more details on using external flash
 
 .. _pm_build_system:
 
+Flash partitions requre device drivers to be enabled for them. The ``DEFAULT_DRIVER_KCONFIG``
+parameter should be set to a Kconfig option that enables the driver during compilation.
```
Comment (FrancescoSer) [inline scripts/partition_manager/partition_manager.rst:534]:
```suggestion
You must enable device drivers for each flash partition. 
To do so, set the ``DEFAULT_DRIVER_KCONFIG`` parameter to the Kconfig option that enables the driver during compilation.
```

Could we be more explicit on indicating here which kconfig option enables the driver during compilation?

**Review by FrancescoSer — CHANGES_REQUESTED** [review]

File: scripts/partition_manager/partition_manager.rst (hunk 1 above)
Comment (de-nordic) [reply scripts/partition_manager/partition_manager.rst:513]: The former.

File: scripts/partition_manager/partition_manager.rst [hunk 5]
```diff
@@ -527,6 +529,13 @@ See :ref:`ug_bootloader_external_flash` for more details on using external flash
 
 .. _pm_build_system:
 
+You must enable device drivers for each flash partition.
+To do so, set the ``DEFAULT_DRIVER_KCONFIG`` parameter to the Kconfig option that enables the driver during compilation.
+When a driver for a partition is not enabled, such partition will not be included into partition manager and will not be accessible at run-time.
```
Comment (FrancescoSer) [inline scripts/partition_manager/partition_manager.rst:534]:
```suggestion
When a driver for a partition is not enabled, such partition will not be included into partition manager and will not be accessible at runtime.
```

File: scripts/partition_manager/partition_manager.rst [hunk 6]
```diff
@@ -527,6 +529,13 @@ See :ref:`ug_bootloader_external_flash` for more details on using external flash
...
+You can use kconfig:option:`CONFIG_PM_OVERRIDE_EXTERNAL_DRIVER_CHECK` to override the behaviour
+and prevent compile-time check for external flash memory devices.
```
Comment (FrancescoSer) [inline scripts/partition_manager/partition_manager.rst:536]:
```suggestion
You can use :kconfig:option:`CONFIG_PM_OVERRIDE_EXTERNAL_DRIVER_CHECK` to override the behaviour and prevent compile-time check for external flash memory devices.
```

File: scripts/partition_manager/partition_manager.rst [hunk 7]
```diff
@@ -527,6 +529,13 @@ See :ref:`ug_bootloader_external_flash` for more details on using external flash
...
+This is useful when the external device does not have a Kconfig option controlled driver.
```
Comment (FrancescoSer) [inline scripts/partition_manager/partition_manager.rst:537]:
```suggestion
This is useful when the external device does not have a driver controlled by kconfig options.
```

**Review by FrancescoSer — CHANGES_REQUESTED** [review]

**Review by FrancescoSer — APPROVED** [review]

File: scripts/partition_manager/partition_manager.rst [hunk 8]
```diff
@@ -527,6 +529,12 @@ See :ref:`ug_bootloader_external_flash` for more details on using external flash
 
 .. _pm_build_system:
 
+You must enable device drivers for each flash partition.
+To do so, set the ``DEFAULT_DRIVER_KCONFIG`` parameter to the Kconfig option that enables the driver during compilation.
+When a driver for a partition is not enabled, such partition will not be included into partition manager and will not be accessible at runtime.
+You can use :kconfig:option:`CONFIG_PM_OVERRIDE_EXTERNAL_DRIVER_CHECK` to override the behaviour and prevent compile-time check for external flash memory devices.
+This is useful when the external device does not have a driver controlled by kconfig options.
```
Comment (peknis) [inline scripts/partition_manager/partition_manager.rst:536]:
```suggestion
You must enable device drivers for each flash partition.
Set the ``DEFAULT_DRIVER_KCONFIG`` parameter to the Kconfig option that enables the driver during compilation.
When a driver for a partition is not enabled, the partition will not be included in the partition manager and will not be accessible at runtime.
You can use the :kconfig:option:`CONFIG_PM_OVERRIDE_EXTERNAL_DRIVER_CHECK` Kconfig option to override the behavior and prevent compile-time check for external flash memory devices.
This is useful when the external device does not have a driver controlled by Kconfig options.
```

**Review by peknis — CHANGES_REQUESTED** [review]

**Review by peknis — APPROVED** [review]

**Pull request merged**